mlserver/env.py

```diff
@@ -0,0 +1,74 @@
+import asyncio
+import os
+import sys
+import tarfile
+import glob
+
+from typing import List
+from functools import cached_property
+
+from .logging import logger
+
+
+def _extract_env(tarball_path: str, env_path: str) -> None:
+    logger.info(f"Extracting environment tarball from {tarball_path}...")
+    with tarfile.open(tarball_path, "r") as tarball:
+        tarball.extractall(path=env_path)
+
+
+class Environment:
+    """
+    Custom Python environment.
+    The class can be used as a context manager to enable / disable the custom
+    environment.
+    """
+
+    def __init__(self, env_path: str):
+        self._env_path = env_path
+
+    @classmethod
+    async def from_tarball(cls, tarball_path: str, env_path: str) -> "Environment":
+        loop = asyncio.get_running_loop()
+        await loop.run_in_executor(None, _extract_env, tarball_path, env_path)
+
+        return cls(env_path)
+
+    @cached_property
+    def sys_path(self) -> List[str]:
```

Comment:
Actually I meant to ask first whether they should be private methods?

Reply:
That's a great point! We probably just need the constructors as public. Everything else can be private, since it will mainly be used as a context manager.